# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationParent.h
# Commit: cc25e5c94813
# Full Hash: cc25e5c94813aac3d7043350f3d1e090e30d2363
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-12 09:32:50
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 1: Add skeleton PNotification r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227686
# ==============================================================================

diff -r af9b668bb659 -r cc25e5c94813 dom/notification/NotificationParent.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/notification/NotificationParent.h	Mon Nov 11 20:34:11 2024 +0000
@@ -0,0 +1,34 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef DOM_NOTIFICATION_NOTIFICATIONPARENT_H_
+#define DOM_NOTIFICATION_NOTIFICATIONPARENT_H_
+
+#include "mozilla/dom/notification/PNotificationParent.h"
+#include "mozilla/ipc/PBackgroundParent.h"
+#include "mozilla/dom/DOMTypes.h"
+
+namespace mozilla::dom::notification {
+
+class NotificationParent final : public PNotificationParent,
+                                 public nsIObserver {
+  using IPCResult = mozilla::ipc::IPCResult;
+
+ public:
+  // Threadsafe to pass the class from PBackground to main thread
+  NS_DECL_THREADSAFE_ISUPPORTS
+  NS_DECL_NSIOBSERVER
+
+  IPCResult RecvShow(ShowResolver&& aResolver);
+  IPCResult RecvClose();
+
+ private:
+  ~NotificationParent() = default;
+};
+
+}  // namespace mozilla::dom::notification
+
+#endif  // DOM_NOTIFICATION_NOTIFICATIONPARENT_H_